src/overlays/actors/ovl_En_Bubble/z_en_bubble.c

```diff
@@ -32,73 +31,399 @@ const ActorInit En_Bubble_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bubble/EnBubble_Draw.s")
+static CollisionCheckInfoInit2 sColChkInfoInit = { 1, 2, 25, 25, MASS_IMMOVABLE };
+
+void EnBubble_SetDimensions(EnBubble* this, f32 dim) {
+    f32 x;
+    f32 y;
+    f32 z;
+    f32 norm;
+
+    this->actor.flags |= ACTOR_FLAG_1;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actor.shape.yOffset = 16.0f;
+    this->modelRotSpeed = 16.0f;
+    this->modelEllipticity = 0.08f;
+    this->modelWidth = dim;
+    this->modelHeight = dim;
+    x = Rand_ZeroOne();
+    y = Rand_ZeroOne();
+    z = Rand_ZeroOne();
+    this->unk_210 = 1.0f;
+    this->unk_214 = 1.0f;
+    norm = SQ(x) + SQ(y) + SQ(z);
+    this->unk1F8.x = x / norm;
+    this->unk1F8.y = y / norm;
+    this->unk1F8.z = z / norm;
+}
+
+u32 func_8089F59C(EnBubble* this) {
+    ColliderInfo* info = &this->colliderSphere.elements[0].info;
+
+    info->toucher.dmgFlags = DMG_EXPLOSIVES;
+    info->toucher.effect = 0;
+    info->toucher.damage = 4;
+    info->toucherFlags = TOUCH_ON;
+    this->actor.velocity.y = 0.0f;
+    return 6;
+}
+
+u32 func_8089F5D0(EnBubble* this) {
+    EnBubble_SetDimensions(this, -1.0f);
+    return 12;
+}
+
+void EnBubble_DamagePlayer(EnBubble* this, PlayState* play) {
+    play->damagePlayer(play, -this->colliderSphere.elements[0].info.toucher.damage);
+    func_800B8E1C(play, &this->actor, 6.0f, this->actor.yawTowardsPlayer, 6.0f);
+}
+
+s32 EnBubble_Explosion(EnBubble* this, PlayState* play) {
+    static Vec3f sEffectAccel = { 0.0f, -0.5f, 0.0f };
+    static Color_RGBA8 sEffectPrimColor = { 255, 255, 255, 255 };
+    static Color_RGBA8 sEffectEnvColor = { 150, 150, 150, 0 };
+    u32 i;
+    Vec3f effectAccel;
+    Vec3f effectVel;
+    Vec3f effectPos;
+
+    effectAccel = sEffectAccel;
+    Math_SmoothStepToF(&this->modelWidth, 4.0f, 0.1f, 1000.0f, 0.0f);
+    Math_SmoothStepToF(&this->modelHeight, 4.0f, 0.1f, 1000.0f, 0.0f);
+    Math_SmoothStepToF(&this->modelRotSpeed, 54.0f, 0.1f, 1000.0f, 0.0f);
+    Math_SmoothStepToF(&this->modelEllipticity, 0.2f, 0.1f, 1000.0f, 0.0f);
+    this->actor.shape.yOffset = ((this->modelHeight + 1.0f) * 16.0f);
+
+    if (DECR(this->explosionCountdown) != 0) {
+        return -1;
+    }
+    effectPos.x = this->actor.world.pos.x;
+    effectPos.y = this->actor.world.pos.y + this->actor.shape.yOffset;
+    effectPos.z = this->actor.world.pos.z;
+    for (i = 0; i < 20; i++) {
+        effectVel.x = (Rand_ZeroOne() - 0.5f) * 7.0f;
+        effectVel.y = Rand_ZeroOne() * 7.0f;
+        effectVel.z = (Rand_ZeroOne() - 0.5f) * 7.0f;
+        EffectSsDtBubble_SpawnCustomColor(play, &effectPos, &effectVel, &effectAccel, &sEffectPrimColor,
+                                          &sEffectEnvColor, Rand_S16Offset(100, 50), 25, 0);
+    }
+    Item_DropCollectibleRandom(play, NULL, &this->actor.world.pos, 0x50);
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    return Rand_S16Offset(90, 60);
+}
+
+u32 func_8089F8BC(EnBubble* this) {
```

Comment:
```suggestion
s32 func_8089F8BC(EnBubble* this) {
```
Especially since you are returning a negative number.

Reply:
Marked as resolved but no change or comment?

Reply:
Something odd went on with my commits, should be good now. Thanks for the catch 👍